### PR TITLE
Add `GraphEdit.cut_nodes_request` signal

### LINKED ISSUE #92776: Visual shader nodes are not deleted with hotkey
### Tested versions

Godot v4.2.2.stable

### System information

macOS 12.6.7 - Vulkan (Mobile) - integrated Apple M1 Pro - Apple M1 Pro (10 Threads)

### Issue description

Graph nodes cannot be deleted with hotkey in visual shader editor. Once delete key is pressed, deletion for scene node is triggered. This behavior is noticed for custom GraphEdit/GraphNode too.

### Steps to reproduce

1. Create new resource -> visual shader
2. Open visual shader editor
3. Create any node
4. Select node created
5. Press delete (Cmd+Backspace)

Expected: graph node is deleted
Actual: pop-up window about deletion of scene node appears

### Minimal reproduction project (MRP)

Empty project

### LINKED ISSUE #97064: Pressing CTRL+X in visual shader editor deletes node in tree rather than cutting out node in visual editor
### Tested versions

- v4.3.stable.mono.official [77dcf97d8]
- v4.2.2.stable.mono.official [15073afe3]

### System information

Godot v4.3.stable.mono - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4080 (NVIDIA; 32.0.15.5599) - AMD Ryzen 9 7950X3D 16-Core Processor (32 Threads)

### Issue description

When I press CTRL+X to cut a node from the visual shader editor and put it into the clipboard instead the currently selected node in the scene tree is cut.

https://github.com/user-attachments/assets/bacd561f-c03b-4ec3-8b5f-efaca70ca8dc

### Steps to reproduce

- Create or open any visual shader, add any node you want to cut.
- Select a node in the scene tree that is not the root. 
- In your visual shader, select a node and press CTRL+X to cut it out.
- Instead of the visual shader node the currently selected node from the scene is cut out.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by aryan-11825114**
Fixes #97064
Should also close #92776

### CODE REVIEW

File: scene/gui/graph_edit.cpp [hunk 1]
```diff
@@ -1999,6 +1999,9 @@ void GraphEdit::gui_input(const Ref<InputEvent> &p_ev) {
 		} else if (p_ev->is_action("ui_copy", true)) {
 			emit_signal(SNAME("copy_nodes_request"));
 			accept_event();
+		} else if (p_ev->is_action("ui_cut")) {
```
Comment (akien-mga) [inline scene/gui/graph_edit.cpp:2002]: Other actions pass `true` to the second parameter here, is there a reason why this one would need to differ?

Comment (aryan-11825114) [reply scene/gui/graph_edit.cpp:2002]: No, it seems to be a mistake on my part, will update it.

Comment (AThousandShips) [reply scene/gui/graph_edit.cpp]: For reference this is the check for exact match

**Review by Geometror — COMMENTED** [review]
Looking good; tested and works as expected!
Last thing remaining: Please squash your commits into one as per https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html

**Comment by Geometror** [comment]
Please adjust the commit message to be more specific: "Add cut_nodes_request signal to GraphEdit"

**Review by Geometror — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**